web/src/components/onboarding/OnboardingTaxId.test.tsx

```diff
@@ -139,18 +162,109 @@ describe("<OnboardingTaxId />", () => {
+          encryptedTaxId: "some-encrypted-value",
+        },
+        errorProps
+      );
+      expect(screen.queryByText(Config.tax.hideButtonText)).not.toBeInTheDocument();
+      fireEvent.click(screen.getByText(Config.tax.showButtonText));
+      expect(mockApi.decryptTaxId).toHaveBeenCalled();
+      await waitFor(() => {
+        expect((screen.getByLabelText("Tax id") as HTMLInputElement).value).toEqual("123-456-789/000");
+      });
+      expect(screen.getByText(Config.tax.hideButtonText)).toBeInTheDocument();
+    });
+
+    it("doesn't decrypt the tax id if tax id is already decrypted", async () => {
+      mockApi.decryptTaxId.mockResolvedValue("123456789000");
+      renderComponent(
+        {
+          ...profileData,
+          taxId: "*******89000",
+          encryptedTaxId: "some-encrypted-value",
+        },
+        errorProps
+      );
+      fireEvent.click(screen.getByText(Config.tax.showButtonText));
+      await waitFor(() => {
+        expect((screen.getByLabelText("Tax id") as HTMLInputElement).value).toEqual("123-456-789/000");
+      });
+      fireEvent.click(screen.getByText(Config.tax.hideButtonText));
+      await waitFor(() => {
+        expect(currentProfileData().taxId).toEqual("123456789000");
+      });
+      fireEvent.click(screen.getByText(Config.tax.showButtonText));
+      await waitFor(() => {
+        expect(mockApi.decryptTaxId).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    it("changes the tax id input type to text when show is clicked and to password when hide is clicked", async () => {
+      mockApi.decryptTaxId.mockResolvedValue("123456789000");
+      renderComponent(
+        {
+          ...profileData,
+          taxId: "********9000",
+          encryptedTaxId: "some-encrypted-value",
+        },
+        errorProps
+      );
+      expect((screen.getByLabelText("Tax id") as HTMLInputElement).type).toEqual("password");
+      fireEvent.click(screen.getByText(Config.tax.showButtonText));
+      await waitFor(() => {
+        expect((screen.getByLabelText("Tax id") as HTMLInputElement).type).toEqual("text");
+      });
+      fireEvent.click(screen.getByText(Config.tax.hideButtonText));
+      await waitFor(() => {
+        expect((screen.getByLabelText("Tax id") as HTMLInputElement).type).toEqual("password");
+      });
+    });
+
+    it("toggles between mobile hide and show text", async () => {
+      mockApi.decryptTaxId.mockResolvedValue("123456789000");
+      setLargeScreen(false);
+      renderComponent(
+        {
+          ...profileData,
+          taxId: "********9000",
+          encryptedTaxId: "some-encrypted-value",
+        },
+        errorProps
+      );
+      expect(screen.queryByText(Config.tax.hideButtonTextMobile)).not.toBeInTheDocument();
+      fireEvent.click(screen.getByText(Config.tax.showButtonTextMobile));
+      await waitFor(() => {
+        expect(screen.getByText(Config.tax.hideButtonTextMobile)).toBeInTheDocument();
```

Comment:
for this and the expect above, I think each one should have a partner expect of what should NOT be in the document. Otherwise, if i just had 2 buttons on the page, one for "show" and one for "hide" and they were just always present, then this test would still pass